realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SessionTests.kt

```diff
@@ -123,9 +124,76 @@ class SessionTests {
         }
     }
 
+    // Check that a Seamless Client Reset is correctly reported.
+    @Test
+    fun errorHandler_discardUnsyncedChangesStrategyReported() = looperThread.runBlocking {
+        val counter = AtomicInteger()
+
+        val incrementAndValidate = {
+            if(2 == counter.incrementAndGet()) {
+                looperThread.testComplete()
+            }
+        }
+
+        val config = SyncConfiguration.Builder(user, "e873fb25-11ef-498f-9782-3c8e1cd2a12c")
+                .assetFile("synced_realm_e873fb25-11ef-498f-9782-3c8e1cd2a12c_no_client_id.realm")
+                .setSyncClientResetStrategy(object: SyncSession.DiscardUnsyncedChangesStrategy{
+                    override fun onBeforeReset(before: Realm, after: Realm) {
+                        Assert.assertEquals(1, before.where<SyncColor>().count())
+                        Assert.assertEquals(0, after.where<SyncColor>().count())
+                        incrementAndValidate()
+                    }
+
+                    override fun onAfterReset(realm: Realm) {
+                        Assert.assertEquals(0, realm.where<SyncColor>().count())
+                        incrementAndValidate()
+                    }
+
+                    override fun onError(session: SyncSession, error: ClientResetRequiredError) {
+                        TODO("Not yet implemented")
+                    }
+
+                })
+                .schema(SyncColor::class.java)
+                .build()
+
+        val realm = Realm.getInstance(config)
+        Assert.assertEquals(1, realm.where<SyncColor>().count())
+        looperThread.closeAfterTest(realm)
+    }
+
+    // Check that a if Seamless loss Client Reset fails the error is correctly reported.
+    @Test
+    fun errorHandler_discardUnsyncedChangesStrategy_resetErrorHandled() = looperThread.runBlocking {
+        val email = TestHelper.getRandomEmail()
+        val user: User = app.registerUserAndLogin(email, "123456")
+
+        val config = configFactory.createSyncConfigurationBuilder(user)
+            .testSchema(SyncStringOnly::class.java)
+                .setSyncClientResetStrategy(object: SyncSession.DiscardUnsyncedChangesStrategy {
+                    override fun onBeforeReset(before: Realm, after: Realm) {
+                        fail("This test case was not supposed to trigger DiscardUnsyncedChangesStrategy::onBeforeReset()")
+                    }
+
+                    override fun onAfterReset(realm: Realm) {
+                        fail("This test case was not supposed to trigger DiscardUnsyncedChangesStrategy::onAfterReset()")
+                    }
+
+                    override fun onError(session: SyncSession, error: ClientResetRequiredError) {
+                        looperThread.testComplete()
```

Comment:
I did not complete this test case, it should be tested as a regular manual client reset.

Reply:
I was mostly wondering if this returns the new 132 error code? I assume that would also contain information about why   it couldn't complete the automatic reset?

Reply:
We will get a Java custom error code 7, with an error message on the line of: `A fatal error occurred during client reset..`. 

Because we are simulating the reset we get the "Simulate Client Reset" error message.